From: Mathieu Baudier <mbaudier@argeo.org>
Date: Sat, 22 Feb 2025 07:52:42 +0100
Subject: improve-cmake-build

---
 src/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 892c233..cef289b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -418,5 +418,9 @@ if (BUILD_SHARED_LIBS)
         set_target_properties(${target} PROPERTIES POSITION_INDEPENDENT_CODE ON)
         target_compile_definitions(${target} PRIVATE GGML_BUILD)
         target_compile_definitions(${target} PUBLIC  GGML_SHARED)
+		if(DEFINED GGML_BUILD_NUMBER)
+	        message(STATUS "Set ${target} shared library version to ${GGML_VERSION_BASE}")
+        	set_target_properties(${target} PROPERTIES VERSION ${GGML_VERSION_BASE} SOVERSION ${GGML_VERSION_MAJOR})
+        endif()
     endforeach()
 endif()
